peer.js

```diff
@@ -410,6 +413,45 @@ TChannelPeer.prototype.connectTo = function connectTo() {
     return self.connect(true);
 };
 
+TChannelPeer.prototype.tryConnect = function tryConnect() {
+    var self = this;
+
+    var rightNow = Date.now();
+    if (rightNow < self.nextConnAttemptTime) {
+        return;
+    }
+
+    var conn = this.getOutConnection();
+    if (!conn || conn.direction !== 'out') {
+        conn = this.connectTo();
+    }
+
+    this.waitForIdentified(conn, onIdentified);
+
+    function onIdentified(err) {
+        if (!err) {
+            self.nextConnAttemptDelay = 0;
+            self.nextConnAttemptTime = 0;
+            return;
+        }
+
+        if (self.nextConnAttemptDelay === 0) {
```

Comment:
Question: since the state about connection attempt delays is stored at the channel level, does this mean that failure to connect to one peer A will also cause backoff from trying to connect to a different peer B?

Wouldn't we want to backoff only in trying to connect to the downed peer and not all peers? Maybe would we could store the delay state at the subchannel level to achieve this?